godotproject/LaserBeam.gd

```diff
@@ -0,0 +1,48 @@
+extends Area2D
+
+enum ANIMATIONS {
+	IDLE, 
+	ATTACKING,
+	MAX_ANIMATIONS,
+}
+
+var AnimationStates = ["Idle", "Attacking"]
+var AnimationState = ANIMATIONS.IDLE
+var inside = false
+var player
+var attackDamage = 1
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	$AnimatedSprite.play(AnimationStates[ANIMATIONS.IDLE])
+	print("Timer started")
+	$Timer.start()
```

Comment:
Can we just set the timer off of oneshot and set it to autostart? Then we can delete $Timer.start() out of this file